### ISSUE TITLE
Instantiation error expected

### ISSUE THREAD

**Issue by flexoron** [infradig/trealla#425]
v1.20.21
```
?- functor(F,a,3), arg(X,F,2), X = 2.
uncaught exception: error(instantiation_error,number)   % surprisingly, other example answers

?- functor(F,a,3), arg(X,F,X), X = 2.
   F = a(2,_A,_B), X = 2   % unexpected
;  F = a(_A,2,_B), X = 2   % expected as such but unexpected due to pmoura's comment
;  false.
```

**Comment by pmoura** [comment]
The instantiation error is **expected** as per the ISO Prolog Core standard for **both** queries. Thus, the second query exposes a bug.

**Comment by pmoura** [comment]
@infradig Pushed additional tests to the Logtalk git version that trigger this bug.

**Comment by infradig** [comment]
Look like was from the early day, trying to emulate a SWI-ism.

**Comment by flexoron** [comment]
fixed with v1.20.22
